Implementer's Guide: Flesh out more details for upward messages (#1556)
paritytech/polkadot: roadmap/implementers-guide/src/messaging.md

that prevent spamming in place.

Upward Message Passing (UMP) is a mechanism responsible for delivering messages in the opposite direction:

> from a parachain up to the relay chain. Upward messagess can serve different purposes and can be of different

Reviewer: ```suggestion
from a parachain up to the relay chain. Upward messages can serve different purposes and can be of different
```

 kinds.

> One kind of messages is `Dispatchable`. They could be thought of similar to extrinsics sent to a relay chain: they also

Reviewer: ```suggestion
One kind of message is `Dispatchable`. They could be thought of similarly to extrinsics sent to a relay chain: they also
```

> One kind of messages is `Dispatchable`. They could be thought of similar to extrinsics sent to a relay chain: they also

Reviewer: Since this is the only kind right now, phrasing as "One kind, and the only current kind, of message..."

invoke exposed runtime entrypoints, they consume weight and require fees. The difference is that they originate from
a parachain. Each parachain has a queue of dispatchables to be executed. There can be only so many dispatchables at a time.
The weight that processing of the dispatchables can consume is limited by a preconfigured value. Therefore, it is possible
that some dispatchables will be left for later blocks. To make the dispatching more fair, the queues are processed turn-by-turn
in a round robin fashion.

> Other kinds of upward messages can be introduced in the future as well. Most likely the HRMP channel managem

Reviewer: unfinished sentence?

Author: 🙈 yeah, I lost a battle with git-rebase there...

> Other kinds of upward messages can be introduced in the future as well. Most likely the HRMP channel management.

Reviewer 2: Was `HRMP` declared before hands?

Author: Thanks, fixed!

> Other kinds of upward messages can be introduced in the future as well. Most likely the HRMP channel management.

Reviewer 2: ```suggestion
Other kinds of upward messages can be introduced in the future as well. Most likely the Horizontal Relay-chain Message Passing (HRMP) channel management.
```

Author: I think this one is outdated

that some dispatchables will be left for later blocks. To make the dispatching more fair, the queues are processed turn-by-turn
in a round robin fashion.

Other kinds of upward messages can be introduced in the future as well. Potential candidates are channel management for

> horizontal message passing (XCMP and HRMP, both are to be described later), new validation code signalling, or other

Reviewer 2: "Later" as in at a later point of time in this document, or as in further down in this document, if so, a link might be helpful

Author: Later, is actually as in later PRs : ) I am preparing one as we speak

Author: Ah sorry I misinterpreted. Further down the document

Author: Addressed
